engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/StorageSystemSnapshotStrategy.java

```diff
@@ -440,41 +526,28 @@ private void markAsBackedUp(SnapshotObject snapshotObj) {
 
     @Override
     public StrategyPriority canHandle(Snapshot snapshot, SnapshotOperation op) {
-        long volumeId = snapshot.getVolumeId();
-        VolumeVO volumeVO = _volumeDao.findById(volumeId);
         if (SnapshotOperation.REVERT.equals(op)) {
-            if (volumeVO != null && ImageFormat.QCOW2.equals(volumeVO.getFormat()))
-                return StrategyPriority.DEFAULT;
-            else
-                return StrategyPriority.CANT_HANDLE;
+            return StrategyPriority.CANT_HANDLE;
         }
 
-        long storagePoolId;
+        long volumeId = snapshot.getVolumeId();
 
-        if (volumeVO == null) {
-            SnapshotDataStoreVO snapshotStore = _snapshotStoreDao.findBySnapshot(snapshot.getId(), DataStoreRole.Primary);
+        VolumeVO volumeVO = _volumeDao.findByIdIncludingRemoved(volumeId);
 
-            if (snapshotStore != null) {
-                storagePoolId = snapshotStore.getDataStoreId();
-            }
-            else {
-                throw new CloudRuntimeException("Unable to determine the storage pool of the snapshot");
-            }
-        }
-        else {
-            storagePoolId = volumeVO.getPoolId();
-        }
+        long storagePoolId = volumeVO.getPoolId();
 
         DataStore dataStore = _dataStoreMgr.getDataStore(storagePoolId, DataStoreRole.Primary);
 
-        Map<String, String> mapCapabilities = dataStore.getDriver().getCapabilities();
+        if (dataStore != null) {
+            Map<String, String> mapCapabilities = dataStore.getDriver().getCapabilities();
 
-        if (mapCapabilities != null) {
-            String value = mapCapabilities.get(DataStoreCapabilities.STORAGE_SYSTEM_SNAPSHOT.toString());
-            Boolean supportsStorageSystemSnapshots = new Boolean(value);
+            if (mapCapabilities != null) {
+                String value = mapCapabilities.get(DataStoreCapabilities.STORAGE_SYSTEM_SNAPSHOT.toString());
+                Boolean supportsStorageSystemSnapshots = new Boolean(value);
```

Comment:
``new Boolean`` skips the constant pool -- placing unnecessary pressure on the garbage collector.  Please use ``Boolean.valueOf`` instead.